### PR TITLE
Adding support for healthchecks to the backend configs.

### PR DESCRIPTION
**Pull request by vbannai**

### CODE REVIEW

**Comment by rramkumar1** [comment]
@bowei PTAL
@skmatti FYI since you have a PR that touches alot of the same files.

File: pkg/apis/backendconfig/v1/types.go [hunk 1]
```diff
@@ -132,3 +133,15 @@ type SessionAffinityConfig struct {
...
+// HealthCheckConfig contains configuration for the health check.
+// +k8s:openapi-gen=true
+type HealthCheckConfig struct {
+	CheckIntervalSec   *int64  `json:"checkIntervalSec,omitempty"`
+	TimeoutSec         *int64  `json:"timeoutSec,omitempty"`
+	HealthyThreshold   *int64  `json:"healthyThreshold,omitempty"`
+	UnhealthyThreshold *int64  `json:"unhealthyThreshold,omitempty"`
+	Type               *string `json:"type,omitempty"`
+	Port               *int64  `json:"port,omitempty"`
+	RequestPath        *string `json:requestPath,omitEmpty"`
```
Comment (AnishShah) [inline pkg/apis/backendconfig/v1/types.go]: This should be `json:"requestPath,omitEmpty"`. Missing inverted comma before requestPath.

**Pull request merged**